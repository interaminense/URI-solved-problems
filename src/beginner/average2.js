/**
 * Read three values (variables A, B and C), which are the three student's grades. Then, calculate
 * the average, considering that grade A has weight 2, grade B has weight 3 and the grade C has
 * eight 5. Consider that each grade can go from 0 to 10.0, always with one decimal place.
 *
 * INPUT
 *
 * The input file contains 3 values of floating points with one digit after the decimal point.
 * @param {number} a
 * @param {number} b
 * @param {number} c
 *
 * OUTPUT
 *
 * Print AVERAGE according to the following example
 */

const tmpl = result => `AVERAGE = ${result}`;

const average = (a, b, c) => tmpl((a * 2 + b * 3 + c * 5) / (2 + 3 + 5));

export { average, tmpl };
