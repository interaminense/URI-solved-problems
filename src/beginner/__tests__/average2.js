import { average, tmpl } from '../average2';

describe('Average 2', () => {
  let a = 0;
  let b = 0;
  let c = 0;

  it('The average for 5, 6 and 7 must be 6.3', () => {
    a = 5;
    b = 6;
    c = 7;

    expect(average(a, b, c)).toEqual(tmpl(6.3));
  });

  it('The average for 5, 10 and 10 must be 9', () => {
    a = 5;
    b = 10;
    c = 10;

    expect(average(a, b, c)).toEqual(tmpl(9));
  });

  it('The average 10, 10 and 5 must be 10', () => {
    a = 10;
    b = 10;
    c = 5;

    expect(average(a, b, c)).toEqual(tmpl(7.5));
  });
});
