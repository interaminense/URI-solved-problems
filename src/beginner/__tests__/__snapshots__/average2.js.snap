// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average 2 The average 10, 10 and 5 must be 10 1`] = `"AVERAGE = 7.5"`;

exports[`Average 2 The average for 5, 6 and 7 must be 6.3 1`] = `"AVERAGE = 6.3"`;

exports[`Average 2 The average for 5, 10 and 10 must be 9 1`] = `"AVERAGE = 9"`;
